md5sum_test.cpp

```diff
@@ -0,0 +1,328 @@
+// MD5 sum--unit test.
+//
+// Copyright (C) 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019 Gregory W. Chicares.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 2 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software Foundation,
+// Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+//
+// http://savannah.nongnu.org/projects/lmi
+// email: <gchicares@sbcglobal.net>
+// snail: Chicares, 186 Belle Woods Drive, Glastonbury CT 06033, USA
+
+#include "pchfile.hpp"
+
+#include "assert_lmi.hpp"
+#include "contains.hpp"
+#include "md5sum.hpp"
+#include "miscellany.hpp"
+#include "system_command.hpp"
+#include "test_tools.hpp"
+
+#include <boost/filesystem/convenience.hpp> // basename()
+#include <boost/filesystem/fstream.hpp>
+#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/path.hpp>
+
+#include <cstdio>
+#include <cstring>                      // memcpy(), strlen()
+#include <fstream>
+#include <string>
+#include <vector>
+
+namespace
+{
+char const* test_filename = "md5_file";
+
+// Use the windows line ending format to test the text mode too.
+std::string const test_text =
+    "The couple bought a lot of vegetables and paid with a cheque. They\r\n"
+    "did some sightseeing, but it was too hot outside. In a park they met\r\n"
+    "a colleague and had a dinner romantically. After the meal they\r\n"
+    "decided to visit the local zoo. A lot of people saw a leopard at the\r\n"
+    "zoo. Near the cage they met the colonel, a well known man doing\r\n"
+    "business. He looked a little bit strange, maybe his ancestors were\r\n"
+    "apache or shoshone indians. They may have travelled from coast to\r\n"
+    "coast and saw interesting places. In the evening every building\r\n"
+    "was lit."
+    ;
+
+char const* test_bin_checksum =  "40e541710871106ebc596595da341dcb";
+
+#if defined LMI_MSW
+char const* test_text_checksum = "e9f0fbd7a758d253ecd48ccded29234c";
+#else
+// Under Linux text and binary modes are the same.
+char const* test_text_checksum = test_bin_checksum;
+#endif // defined LMI_MSW
+
+char const* md5sums_filename = "md5_sums";
+
+std::string const md5sums_text =
+    "00112233445566778899aabbccddeeff  test\n"
+    "ffeeddccbbaa99887766554433221100 *nix\n"
```

Comment:
Not sure why has this changes from the previously used `test2`, but in any case, I'd like to rename the file name used there to `test.bin` and, for consistency, also rename the one in the line above to `test.txt`. This should get rid of `make check_concinnity` warnings on this line (but please check that it does). TIA!